src/cpuinfo/SDL_cpuinfo.c

```diff
@@ -840,18 +844,40 @@ static const char *SDL_GetCPUName(void)
 int SDL_GetCPUCacheLineSize(void)
 {
     const char *cpuType = SDL_GetCPUType();
+    int cacheline_size;
     int a, b, c, d;
     (void)a;
     (void)b;
     (void)c;
     (void)d;
     if (SDL_strcmp(cpuType, "GenuineIntel") == 0 || SDL_strcmp(cpuType, "CentaurHauls") == 0 || SDL_strcmp(cpuType, "  Shanghai  ") == 0) {
         cpuid(0x00000001, a, b, c, d);
-        return ((b >> 8) & 0xff) * 8;
+	cacheline_size = ((b >> 8) & 0xff) * 8;
+        return cacheline_size;
     } else if (SDL_strcmp(cpuType, "AuthenticAMD") == 0 || SDL_strcmp(cpuType, "HygonGenuine") == 0) {
         cpuid(0x80000005, a, b, c, d);
-        return c & 0xff;
+	cacheline_size = c & 0xff;
+        return cacheline_size;
     } else {
+#if defined(__LINUX__) && defined (__GLIBC__)
+        if ((cacheline_size = sysconf(_SC_LEVEL1_DCACHE_LINESIZE)) != -1)
+            return cacheline_size;
+#endif
+#if defined(__LINUX__) /* non glibc linux systems */
+        FILE *f = NULL;
+
+        f = fopen("/sys/devices/system/cpu/cpu0/cache/index0/coherency_line_size", "r");
+        if (f) {
+            fscanf(f, "%d", &cacheline_size);
```

Comment:
CI runs fail because of unchecked result:

```
/home/runner/work/SDL/SDL/src/cpuinfo/SDL_cpuinfo.c: In function ‘SDL_GetCPUCacheLineSize_REAL’:
/home/runner/work/SDL/SDL/src/cpuinfo/SDL_cpuinfo.c:871:13: error: ignoring return value of ‘fscanf’ declared with attribute ‘warn_unused_result’ [-Werror=unused-result]
  871 |             fscanf(f, "%d", &cacheline_size);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
Something like this should do the trick:

```diff
diff --git a/src/cpuinfo/SDL_cpuinfo.c b/src/cpuinfo/SDL_cpuinfo.c
index 742988c..b57d595 100644
--- a/src/cpuinfo/SDL_cpuinfo.c
+++ b/src/cpuinfo/SDL_cpuinfo.c
@@ -869,7 +869,7 @@ int SDL_GetCPUCacheLineSize(void)
         f = fopen("/sys/devices/system/cpu/cpu0/cache/index0/coherency_line_size", "r");
         if (f) {
-            fscanf(f, "%d", &cacheline_size);
+            int r = fscanf(f, "%d", &cacheline_size);
             fclose(f);
-            return cacheline_size;
+            return (r == 1) ? cacheline_size : SDL_CACHELINE_SIZE;
 	}
 #elif defined __FreeBSD__
```

Reply:
Oh sorry didn't saw that, going to push another commit with that diff. Was fighting with emacs to convert tabs to spaces.

Reply:
Just pushed a commit with that diff. Can you please verify with CI?